### DISCUSSION TITLE
retroactive camel and Pascal case

### DISCUSSION THREAD

**Discussion by Josiah-tan** [opensteno/plover#1581] [General]
for snake case, I can use retro_everything to perform the following
"STPHA*EUBGD": "=retro_everything:{MODE:SNAKE},{MODE:RESET}"
`STREUPBG/O/STPHAEUBG/STPHA*EUBGD/STPHA*EUBGD` → string_to_snake

however, for camel case this doesn't work
"KA*FRLD": "=retro_everything:{MODE:CAMEL},{MODE:RESET}",
`STREUPBG/O/KAFRL/KA*FRLD/KA*FRLD` → stringtoCamel

perhaps an idea is to implement Pascal case and use that with retro_everything to fix the problem

**Comment by user202729** [comment]
```
{}{:retro_stringop:7:__import__('re').sub(r"\s+(\S+)\s*$", lambda match: match[1][0].upper()+match[1][1:], text)}
```

I use this one but more often than not I actually want title case and end up doing it the long way again.

(the code is unreadable though)

**Reply by Josiah-tan** [reply]
This is brilliant thanks for the idea! Didn't know that plover-retro-stringop existed!
To run the code above:

install plover-retro-stringop from the registry
or
`pip install plover-retro-stringop`

The entries I put into my dictionary:
```
"KA*FRLD": "{}{:retro_stringop:7:__import__('re').sub(r\"\\s+(\\w+)\\s*$\", lambda match: match[1][0].upper()+match[1][1:], text)}",
"SKPA*LD": "{}{:retro_stringop:7:__import__('re').sub(r\"(\\w+)\\s+(\\w+)\\s*$\", lambda match: match.groups()[0][0].upper() + match.groups()[0][1:] + match.groups()[1][0].upper() + match.groups()[1][1:], text)\n}",
```
First one is `camelled`, second one is `Pascalled`

(edit: I think `\w` might make more sense than `\S` unless there is something I am missing here?)
